engine/runtime/src/main/java/org/enso/interpreter/runtime/number/EnsoBigInteger.java

```diff
@@ -1,23 +1,24 @@
 package org.enso.interpreter.runtime.number;
 
+import java.math.BigInteger;
+
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.library.dispatch.TypesLibrary;
+
 import com.oracle.truffle.api.CompilerDirectives;
 import com.oracle.truffle.api.interop.InteropLibrary;
 import com.oracle.truffle.api.interop.TruffleObject;
 import com.oracle.truffle.api.interop.UnsupportedMessageException;
 import com.oracle.truffle.api.library.CachedLibrary;
 import com.oracle.truffle.api.library.ExportLibrary;
 import com.oracle.truffle.api.library.ExportMessage;
-import org.enso.interpreter.runtime.EnsoContext;
-import org.enso.interpreter.runtime.data.Type;
-import org.enso.interpreter.runtime.library.dispatch.TypesLibrary;
-
-import java.math.BigInteger;
 
 /** Internal wrapper for a {@link BigInteger}. */
 @ExportLibrary(InteropLibrary.class)
 @ExportLibrary(TypesLibrary.class)
 @SuppressWarnings("truffle-abstract-export")
-public final class EnsoBigInteger extends Number implements TruffleObject {
+public final class EnsoBigInteger implements TruffleObject {
```

Comment:
Why is it no longer a Number?

Reply:
That was a cheat which can cause more problems than it used to solve. With the introduction of `asBigInteger()` message, such cheating is no longer necessary.